pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdConnectors.java

```diff
@@ -0,0 +1,415 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.admin.cli;
+
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import com.beust.jcommander.converters.StringConverter;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.pulsar.client.admin.PulsarAdmin;
+import org.apache.pulsar.functions.shaded.proto.Function.FunctionDetails;
+import org.apache.pulsar.functions.utils.SinkConfig;
+import org.apache.pulsar.functions.utils.SourceConfig;
+
+import java.io.File;
+import java.io.IOException;
+
+@Slf4j
+@Getter
+@Parameters(commandDescription = "Interface for managing Pulsar Connectors (Ingress and egress data to and from " +
+        "Pulsar)")
+public class CmdConnectors extends CmdBase {
+
+    private final CreateSource createSource;
+    private final CreateSink createSink;
+    private final DeleteConnector deleteConnector;
+
+    public CmdConnectors(PulsarAdmin admin) {
+        super("connectors", admin);
+        createSource = new CreateSource();
+        createSink = new CreateSink();
+        deleteConnector = new DeleteConnector();
+        jcommander.addCommand("create-source", createSource);
+        jcommander.addCommand("create-sink", createSink);
+        jcommander.addCommand("delete", deleteConnector);
+    }
+
+    /**
+     * Base command
+     */
+    @Getter
+    abstract class BaseCommand extends CliCommand {
+        @Override
+        void run() throws Exception {
+            processArguments();
+            runCmd();
+        }
+
+        void processArguments() throws Exception {
+        }
+
+        abstract void runCmd() throws Exception;
+    }
+
+    @Parameters(commandDescription = "Create Pulsar source connectors")
+    class CreateSource extends BaseCommand {
+        @Parameter(names = "--fqfn", description = "The Fully Qualified Function Name (FQFN) for the function")
```

Comment:
Can we remove this and instead assume for now that all connectors are running in public/default tenant/namespace? And then based on how Oath wants to run it revisit this question later. IMO this exposes too much the function underpinnings of the connect framework